tests/python/unittest/test_ndarray.py

```diff
@@ -1495,6 +1495,45 @@ def test_dlpack():
             mx.test_utils.assert_almost_equal(a_np, d_np)
             mx.test_utils.assert_almost_equal(a_np, e_np)
 
+@with_seed()
+def test_ndarray_is_inf():
+    random_dimensions = np.random.randint(3, 5)
+    random_shape = [np.random.randint(2, 5) for i in range(random_dimensions)]
+    data = mxnet.test_utils.rand_ndarray(random_shape,'default')
+    data[0][0][0]=np.inf
+    data[0][0][1]=-np.inf
+    data[0][1][0]=np.nan
+    output = mx.nd.contrib.isinf(data)
+    expected_output = np.isinf(data.asnumpy())
+    assert_almost_equal(output.asnumpy(), expected_output.astype(int))
+    # astype since numpy functions default return type is boolean array instead of int
+
+@with_seed()
+def test_ndarray_is_finite():
+    random_dimensions = np.random.randint(3, 5)
+    random_shape = [np.random.randint(2, 5) for i in range(random_dimensions)]
+    data = mxnet.test_utils.rand_ndarray(random_shape,'default')
+    data[0][0][0]=np.inf
+    data[0][0][1]=-np.inf
+    data[0][1][0]=np.nan
+    output = mx.nd.contrib.isfinite(data)
+    expected_output = np.isfinite(data.asnumpy())
+    assert_almost_equal(output.asnumpy(), expected_output.astype(int))
+    # astype since numpy functions default return type is boolean array instead of int
+
+@with_seed()
+def test_ndarray_is_nan():
+    random_dimensions = np.random.randint(3, 5)
+    random_shape = [np.random.randint(2, 5) for i in range(random_dimensions)]
+    data = mxnet.test_utils.rand_ndarray(random_shape,'default')
+    data[0][0][0]=np.inf
+    data[0][0][1]=-np.inf
+    data[0][1][0]=np.nan
+    output = mx.nd.contrib.isnan(data)
+    expected_output = np.isnan(data.asnumpy())
+    assert_almost_equal(output.asnumpy(), expected_output.astype(int))
```

Comment:
this should be assert_equal not assert_almost_equal

Reply:
Makes sense. Since I am passing the same data, it is necessary that it's exactly equal (not "almost")
Thanks for pointing that out.